web/src/compose_setup.js

```diff
@@ -180,9 +181,24 @@ export function initialize() {
                     // user-triggered edit to that field.
                     $input.trigger("input");
 
-                    // TODO: Probably this should also renarrow to the
-                    // new topic, if we were currently viewing the old
-                    // topic, just as if a message edit had occurred.
+                    // renarrow to the unresolved topic if currently viewing the resolved topic
+                    const terms = narrow_state.search_terms();
+                    const currently_viewing_channel = terms.find(
+                        (term) => term.operator === "channel",
+                    )?.operand;
+                    const currently_viewing_topic = terms.find(
+                        (term) => term.operator === "topic",
+                    )?.operand;
```

Comment:
`narrow_state.current_filter` is a better API for this; `filter.has_topic()` and `filter.operands("topic")` can get you these values...

Reply:
`filter.has_topic()` simplifies things a lot. 
I could not find `narrow_state.current_filter()` as you had earlier suggested (I think you were referring to the `narrow_state.filter()` method), and therefore used `search_terms()`, which made use of the `filter` method.